tests/build/image_controller.go

```diff
@@ -0,0 +1,276 @@
+							if err != nil {
+								GinkgoWriter.Printf("failed to get logs for pipelinerun %s: %+v\n", pipelineRun.Name, err)
+							}
+							Fail(failMessage)
+						}
+					}
+					return true
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for the PipelineRun to finish")
+			})
+			It("image repo and robot account created successfully", func() {
+
+				component, err := f.AsKubeAdmin.HasController.GetHasComponent(componentName, testNamespace)
+				Expect(err).ShouldNot(HaveOccurred(), "could not get component")
+
+				annotations := component.GetAnnotations()
+				imageRepoName, err = build.GetQuayImageName(annotations)
+				Expect(err).ShouldNot(HaveOccurred(), "failed to read image name")
+
+				imageExist, err := build.DoesImageRepoExistInQuay(imageRepoName)
+				Expect(err).ShouldNot(HaveOccurred(), "failed while checking if image repo exists in quay")
+				Expect(imageExist).To(BeTrue(), "quay image does not exists")
+
+				robotAccountName = build.GetRobotAccountName(imageRepoName)
+				robotAccountExist, err := build.DoesRobotAccountExistInQuay(robotAccountName)
+				Expect(err).ShouldNot(HaveOccurred(), "failed while checking if robot account exists in quay")
+				Expect(robotAccountExist).To(BeTrue(), "quay robot account does not exists")
+
+			})
+			It("image tag is updated successfully", func() {
+				//TODO: check if image tag present once below issue is resolved
+				// https://issues.redhat.com/browse/SRVKP-3064
+				// https://github.com/openshift-pipelines/pipeline-service/pull/632
+
+			})
+			It("related image repo and robot account should be deleted after deleting the component", func() {
+				timeout = time.Second * 60
+				interval = time.Second * 1
+				Expect(f.AsKubeAdmin.HasController.DeleteHasComponent(componentName, testNamespace, true)).To(Succeed())
+				// Check removal of image repo
+				Eventually(func() bool {
+					imageExist, err := build.DoesImageRepoExistInQuay(imageRepoName)
+					return err == nil && !imageExist
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for image repo to be deleted")
+				// Check removal of robot account
+				Eventually(func() bool {
+					robotAccountExist, err := build.DoesRobotAccountExistInQuay(robotAccountName)
+					return err == nil && !robotAccountExist
+				}, timeout, interval).Should(BeTrue(), "timed out when waiting for robot account to be deleted")
+
+			})
+		})
+		When("a component with image controller annotation delete-repo=false is created", func() {
+			BeforeAll(func() {
+				deleteRepo := false
+				// Create a component with Git Source URL and a specified git branch
+				_, err = f.AsKubeAdmin.HasController.CreateComponentWithPaCEnabled(applicationName, componentName, testNamespace, imageControllerComponentGitSourceURL, componentBaseBranchName, deleteRepo)
+				Expect(err).ShouldNot(HaveOccurred())
+			})
+			It("triggers a PipelineRun", func() {
```

Comment:
there are couple of specs very similar to the ones in pac-webhook-build test. the jira ticket mentions to rather modify the existing e2e test - wouldn't that be more effective approach?

thing is that this test creates another pipelinerun that will run in parallel with other tests in CI, which generates more load on the CI cluster that might lead to even lower stability of e2e-tests (we already saw this happening in the past)

Reply:
Agree that, it will create one more pipelinerun since it will run in parallel with other tests, was not aware that it was causing stability issue in the e2e-tests. With my approach I tried to keep image-controller tests in different module for better maintainability and also to reduce the total execution time. Since we are facing stability issue more and total execution time is less of concern, then it would make sense to merge these tests with current build test suite.